hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileSystemImpl.java

```diff
@@ -82,17 +82,8 @@ protected boolean mkdir(String path) {
   @Override
   protected boolean rename(String src, String dst) {
     try {
-      if (fs instanceof DistributedFileSystem) {
-        DistributedFileSystem dfs = (DistributedFileSystem)fs;
-        dfs.rename(new Path(src), new Path(dst), Options.Rename.OVERWRITE);
-        return true;
-      } else {
-        // Replace should be atomic but not available
-        if (fs.exists(new Path(dst))) {
-          fs.delete(new Path(dst), true);
-        }
-        return fs.rename(new Path(src), new Path(dst));
-      }
+      FileUtil.rename(fs, new Path(src), new Path(dst), Options.Rename.OVERWRITE);
```

Comment:
Hi @simbadzina , can you explain why we changed it here? Thanks.

Reply:
Hi @tomscut , if we aren't using a DistributedFileSystem, the old code does the replace itself by using a delete (if necessary) and then rename. This code is already available in Filesystem.java.

More importantly, if there is a non DistributedFileSystem that has an atomic rename, the old code prevents us from using the available atomic rename.

My change delegates the implementation of rename to the underlying Filesystem.
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L1648-L1650

Reply:
I see. Thanks @simbadzina for your explanation.